File: sdk/nodejs/src/model/record.ts

export interface Record {
    id: string
    properties: object
    packedProperties?: object[]
}

export const RecordEntityInfo = {
    namespace: "system",
    resource: "Record",
    restPath: "system-record",
}

export const RecordResource = {
  "name": "Record",
  "namespace": {
    "name": "system"
  },
  "virtual": true,
  "properties": {
    "id": {
      "type": "UUID",
      "required": true,
      "immutable": true,
      "exampleValue": "a39621a4-6d48-11ee-b962-0242ac120002",
      "description": "The unique identifier of the resource. It is randomly generated and immutable.",
      "annotations": {
        "PrimaryProperty": "true",
        "SpecialProperty": "true"
      }
    },
    "packedProperties": {
      "type": "LIST",
      "item": {
        "type": "OBJECT"
      },
      "annotations": {
        "OpenApiHide": "true"
      }
    },
    "properties": {
      "type": "OBJECT",
      "required": true,
      "title": "Properties",
      "description": "The properties of the record. The schema of properties are defined in the resource definition. \nHere you will put the payload corresponding to the resource definition.\n"
    }
  },
  "title": "Generic Record",
  "description": "A generic record resource. All Apis are extended from Generic Record resource",
  "annotations": {
    "RestApiDisabled": "true"
  }
} as unknown

File: pkg/resources/record.go
package resources

import (
	"github.com/apibrew/apibrew/pkg/model"
	"github.com/apibrew/apibrew/pkg/resources/special"
	"github.com/apibrew/apibrew/pkg/service/annotations"
	"github.com/apibrew/apibrew/pkg/util"
)

var RecordResource = &model.Resource{
	Name:        "Record",
	Title:       util.Pointer("Generic Record"),
	Description: util.Pointer("A generic record resource. All Apis are extended from Generic Record resource"),
	Namespace:   "system",
	Virtual:     true,
	Properties: map[string]*model.ResourceProperty{
		"id": special.IdProperty,
		"properties": {
			Title: util.Pointer("Properties"),
			Description: util.Pointer(`The properties of the record. The schema of properties are defined in the resource definition. 
Here you will put the payload corresponding to the resource definition.
`),
			Type:     model.ResourceProperty_OBJECT,
			Required: true,
		},
		"packedProperties": {
			Type: model.ResourceProperty_LIST,
			Item: &model.ResourceProperty{
				Type: model.ResourceProperty_OBJECT,
			},
			Annotations: map[string]string{
				annotations.OpenApiHide: "true",
			},
		},
	},

	Annotations: map[string]string{
		//annotations.SelfContainedProperty: "properties",
		annotations.RestApiDisabled: annotations.Enabled,
	},
}
